proto/hummock.proto

```diff
@@ -358,6 +358,10 @@ message CompactTask {
     TRACK_SST_OBJECT_ID_FAILED = 12;
     NO_AVAIL_CPU_RESOURCE_CANCELED = 13;
     HEARTBEAT_PROGRESS_CANCELED = 14;
+
+    // for serverless compaction
+    SERVERLESS_SEND_FAIL_CANCELED = 15;
```

Comment:
Will serverless compaction controller add more error codes in the future? I am thinking whether we should have one error code only for all serverless compaction controller error.

Reply:
> Will serverless compaction controller add more error codes in the future?

I think it's possible, but not in a near future since the serverless protocol won't change frequently.

> I am thinking whether we should have one error code only for all serverless compaction controller error.

I'm fine if the error message isn't lost. But seems we rely on the error code to tell the error kind, I think we'd better add more when necessary.

Reply:
I think it depends on whether we introduce more state to the compact task, e.g. table_catalog. As we iterate through the logic of the compact task, we may introduce more corresponding error